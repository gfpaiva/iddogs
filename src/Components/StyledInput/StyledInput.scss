@import '../../Scss/root';

.styled-input {
  border-bottom: 1px solid $gray;
  display: block;
  margin: 0 auto;
  max-width: 350px;
  position: relative;
  width: 100%;

  &:after,
  &:before {
    background-color: $secondary-color;
    bottom: -1px;
    content: '';
    display: block;
    height: 1px;
    position: absolute;
    transition: width .5s ease;
    width: 0;
  }

  &:after { left: 50% }
  &:before { right: 50% }

  &--active {
    &:after,
    &:before { width: 50% }
  }

  &--error {
    &:after,
    &:before { background-color: $error-color }
  }

  input {
      background-color: transparent;
      border: none;
      outline: none;
      padding: .5em 1em;
      width: 100%;
    }
}
