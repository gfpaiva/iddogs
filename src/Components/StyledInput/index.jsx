import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import './StyledInput.scss';

export default function StyledInput({
  id,
  className,
  filled,
  error,
  ...props
}) {
  const [active, setActive] = useState(false);

  useEffect(() => {
    if (!active) setActive(filled);
  }, [filled]);

  return (
    <label className={`styled-input${active ? ' styled-input--active' : ''}${error ? ' styled-input--error' : ''}`} htmlFor={id}>
      <input
        {...props}
        id={id}
        className={`text-center${className ? ` ${className}` : ''}`}
        onFocus={() => setActive(true)}
        onBlur={() => !filled && setActive(false)}
      />
    </label>
  );
}

StyledInput.defaultProps = {
  className: '',
  error: false,
  filled: false,
};

StyledInput.propTypes = {
  className: PropTypes.string,
  error: PropTypes.bool,
  filled: PropTypes.bool,
  id: PropTypes.string.isRequired,
};
