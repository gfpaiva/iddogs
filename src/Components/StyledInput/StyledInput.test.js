import React from 'react';
import { mount } from 'enzyme';

import StyledInput from './index';

describe('<StyledInput />', () => {
  it('should mount properly', () => {
    const wrapper = mount(<StyledInput id="test" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should style properly on focus and blur', () => {
    const wrapper = mount(<StyledInput id="test" />);

    wrapper.find('input').simulate('focus');

    expect(wrapper.find('label').hasClass('styled-input--active')).toBeTruthy();

    wrapper.find('input').simulate('blur');

    expect(wrapper.find('label').hasClass('styled-input--active')).toBeFalsy();
  });

  it('should style properly on filled', () => {
    const wrapper = mount(<StyledInput id="test" filled />);

    expect(wrapper.find('label').hasClass('styled-input--active')).toBeTruthy();
  });

  it('should style properly on error', () => {
    const wrapper = mount(<StyledInput id="test" error />);

    expect(wrapper.find('label').hasClass('styled-input--error')).toBeTruthy();
  });
});
