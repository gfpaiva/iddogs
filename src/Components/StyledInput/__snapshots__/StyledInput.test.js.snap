// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledInput /> should mount properly 1`] = `
<StyledInput
  className=""
  error={false}
  filled={false}
  id="test"
>
  <label
    className="styled-input"
    htmlFor="test"
  >
    <input
      className="text-center"
      id="test"
      onBlur={[Function]}
      onFocus={[Function]}
    />
  </label>
</StyledInput>
`;
